File: Assets/Scripts/ArrowBow/Trigger.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class Trigger : MonoBehaviour {

    public bool activated = false;
	
	void OnTriggerStay(Collider other) {
        if (activated)
        {
            if (other.name == "Player" && Input.GetKey(KeyCode.E))
            {
                other.gameObject.BroadcastMessage("PickUpArrow");
                Destroy(transform.root.gameObject);
            }
        }
    }
}

File: Assets/Resources/Scripts/Trigger.js
#pragma strict 

var activated : boolean = false;
function OnTriggerStay (other : Collider) {
	
	if(activated){
		if(other.name == "Player" && Input.GetKey(KeyCode.E)){
			other.gameObject.BroadcastMessage ("PickUpArrow");
			Destroy(transform.root.gameObject);
		}
	}
}
	